src/IntelliHR/Validation/Validators/MaxDate.php

```diff
@@ -2,65 +2,51 @@
 
 namespace IntelliHR\Validation\Validators;
 
-use DateTime;
+use Exception;
 use Illuminate\Contracts\Validation\Validator;
-use InvalidArgumentException;
 
 class MaxDate extends AbstractValidator
 {
     /**
-     * Name of the validator
-     *
      * @var string
      */
     public static $name = 'max_date';
 
     /**
-     * Fallback message
-     *
      * @var string
      */
     public static $message = ':attribute must be before :max_date';
 
-    /**
-     * @param           $attribute
-     * @param           $value
-     * @param array     $parameters
-     * @param Validator $validator
-     *
-     * @return bool
-     * @throws InvalidArgumentException
-     */
     public function validateMaxDate(
         $attribute,
         $value,
         array $parameters,
         Validator $validator
-    ) {
+    ): bool {
         $this->requireParameterCount(1, $parameters, self::$name);
+        $format = $parameters[1] ?? null;
+
+        $maxDate = $this->getDateForParameter($parameters[0], $format);
```

Comment:
Would this also need to be in the try/catch?

Reply:
I want invalid parameters to still throw an exception, as that's a code issue rather than a validation issue

Reply:
Fair enough. This behaviour is tested from memory